tests/gbs/test_subgraph.py

```diff
@@ -14,113 +14,239 @@
 r"""
 Unit tests for strawberryfields.gbs.subgraph
 """
-# pylint: disable=no-self-use,unused-argument
+# pylint: disable=no-self-use,unused-argument,protected-access,redefined-outer-name
 import functools
 
 import networkx as nx
 import numpy as np
 import pytest
 
-from strawberryfields.gbs import sample, subgraph
+from strawberryfields.gbs import data, subgraph
 
 pytestmark = pytest.mark.gbs
 
+p_planted = data.Planted()
+g_planted = nx.Graph(p_planted.adj)
 
-samples_subgraphs = np.array(
-    [
-        [1, 2, 3, 4],
-        [0, 1, 2, 3],
-        [0, 1, 2, 3],
-        [0, 1, 2, 5],
-        [0, 1, 2, 3],
-        [0, 1, 4, 5],
-        [0, 1, 4, 5],
-        [0, 1, 3, 4],
-        [0, 1, 2, 3],
-        [0, 1, 4, 5],
-    ]
-)
+
+@pytest.fixture()
+def process_planted(min_size, max_size, max_count, n_samples):
+    """Fixture for loading samples from the Planted dataset"""
+    samples = p_planted[:n_samples]
+    samples.insert(10, [31])  # include invalid subgraph to test that function simply continues
```

Comment:
I don't understand the role of this insert. Is it a test within a fixture? What happens if you remove this line?

Reply:
Good catch! The function was set to continue if an invalid subgraph was presented (i.e., one with nodes not in the actual subgraph, such as the one above). In the end I decided this was an unnecessary complication and took this out.